app/services/queries/cclow/litigation_query.rb

```diff
@@ -78,106 +77,20 @@ def filter_by_to_status
-      end
-
-      def filter_by_litigation_side_c
-        return scope unless params[:side_c].present?
-
-        scope.includes(:litigation_sides)
-          .where(litigation_sides: {side_type: 'c', name: params[:side_c]})
-      end
-
-      def filter_by_litigation_side_a_b
-        return scope unless params[:side_a].present? && params[:side_b].present?
+        sql = ''
 
-        sql = "
-          SELECT side_a.litigation_id
-          FROM litigation_sides side_a, litigation_sides side_b
-          WHERE side_a.side_type = 'a' AND side_a.name IN (#{params[:side_a].map { |a| "'#{a}'" }.join(', ')})
-          AND side_b.side_type = 'b' AND side_b.name IN (#{params[:side_b].map { |b| "'#{b}'" }.join(', ')})
-          AND side_a.litigation_id = side_b.litigation_id;
-        "
+        [:side_a, :side_b, :side_c].each do |side|
+          next unless params[side].present?
 
-        litigation_resources(sql)
-      end
-
-      def filter_by_litigation_side_a_b_c
-        return scope unless params[:side_a].present? && params[:side_b].present? && params[:side_c].present?
-
-        sql = "
-          SELECT side_a.litigation_id
-          FROM litigation_sides side_a, litigation_sides side_b, litigation_sides side_c
-          WHERE side_a.side_type = 'a' AND side_a.name IN (#{params[:side_a].map { |a| "'#{a}'" }.join(', ')})
-          AND side_b.side_type = 'b' AND side_b.name IN (#{params[:side_b].map { |b| "'#{b}'" }.join(', ')})
-          AND side_c.side_type = 'c' AND side_c.name IN (#{params[:side_c].map { |c| "'#{c}'" }.join(', ')})
-          AND side_a.litigation_id = side_b.litigation_id
-          AND side_a.litigation_id = side_c.litigation_id;
-        "
-
-        litigation_resources(sql)
-      end
-
-      def filter_by_litigation_side_b_c
-        return scope unless params[:side_b].present? && params[:side_c].present?
-
-        sql = "
-          SELECT side_b.litigation_id
-          FROM litigation_sides side_b, litigation_sides side_c
-          WHERE side_b.side_type = 'b' AND side_b.name IN (#{params[:side_b].map { |b| "'#{b}'" }.join(', ')})
-          AND side_c.side_type = 'c' AND side_c.name IN (#{params[:side_c].map { |c| "'#{c}'" }.join(', ')})
-          AND side_b.litigation_id = side_c.litigation_id;
-        "
-
-        litigation_resources(sql)
-      end
-
-      def filter_by_litigation_side_a_c
-        return scope unless params[:side_a].present? && params[:side_c].present?
-
-        sql = "
-          SELECT side_a.litigation_id
-          FROM litigation_sides side_a, litigation_sides side_c
-          WHERE side_a.side_type = 'a' AND side_a.name IN (#{params[:side_a].map { |a| "'#{a}'" }.join(', ')})
-          AND side_c.side_type = 'c' AND side_c.name IN (#{params[:side_c].map { |c| "'#{c}'" }.join(', ')})
-          AND side_a.litigation_id = side_c.litigation_id;
-        "
+          sql += <<-SQL
+            INNER JOIN litigation_sides AS #{side}
```

Comment:
that's slick 🤔and I was full on with trying to invent ruby's pattern matching for that case 😅